File: Python/Two Sum II - Input array is sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for index, num in enumerate(numbers):
            if target - num in dic:
                return [dic[target - num], index + 1]
            else:
                dic[num] = index + 1
            

File: cpp/Two Sum II - Input array is sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        map<int, int> dict;
        vector<int> result;
        for (int i = 0; i < numbers.size(); ++i) {
            if (dict.find(target - numbers[i]) != dict.end()) {
                result.push_back(dict[target- numbers[i]]);
                result.push_back(i + 1);
                break;
            } else {
                dict[numbers[i]] = i + 1;
            }
        }
        return result;
    }
};
